### PR TITLE
Simplify/improve barrier in AllReduce6

### PR DESCRIPTION
**Pull request by roshandathathri**
Drop superfluous __threadfence_system()

### CODE REVIEW

File: python/mscclpp_benchmark/allreduce.cu [hunk 1]
```diff
@@ -788,6 +788,24 @@ extern "C" __global__ void __launch_bounds__(1024, 1)
 
 #if defined(__CUDA_ARCH__) && __CUDA_ARCH__ >= 900
 
+// Barrier among all devices followed by a memory fence
+// Should be called by all threads on all devices
+// Assumes \p num_threads_per_block >= \p num_ranks
+__device__ void barrier(mscclpp::SmDevice2DeviceSemaphoreDeviceHandle* semaphores, int thread_id, int block_id,
```
Comment (Binyang2014) [inline python/mscclpp_benchmark/allreduce.cu:794]: make it `__forceinline__`?

File: python/mscclpp_benchmark/allreduce.cu [hunk 2]
```diff
@@ -788,6 +788,24 @@ extern "C" __global__ void __launch_bounds__(1024, 1)
...
+                        int num_threads_per_block, int num_blocks, int num_ranks) {
```
Comment (Binyang2014) [inline python/mscclpp_benchmark/allreduce.cu:795]: Seems `num_threads_per_block` is not used

Comment (roshandathathri) [reply python/mscclpp_benchmark/allreduce.cu:795]: It is not

File: python/mscclpp_benchmark/allreduce.cu (hunk 1 above)
Comment (roshandathathri) [reply python/mscclpp_benchmark/allreduce.cu:794]: Will do

Comment (roshandathathri) [reply python/mscclpp_benchmark/allreduce.cu]: Pushed the changes

**Review by chhwang — APPROVED** [review]
LGTM

**Pull request merged**